inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/InstrumentationUpdateService.java

```diff
@@ -0,0 +1,176 @@
+package rocks.inspectit.oce.core.instrumentation;
+
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import net.bytebuddy.description.type.TypeDescription;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.event.EventListener;
+import org.springframework.stereotype.Service;
+import rocks.inspectit.oce.core.config.InspectitConfigChangedEvent;
+import rocks.inspectit.oce.core.config.InspectitEnvironment;
+import rocks.inspectit.oce.core.config.model.instrumentation.InstrumentationSettings;
+import rocks.inspectit.oce.core.config.model.instrumentation.InternalSettings;
+import rocks.inspectit.oce.core.instrumentation.special.SpecialSensor;
+import rocks.inspectit.oce.core.selfmonitoring.SelfMonitoringService;
+import rocks.inspectit.oce.core.service.BatchJobExecutorService;
+import rocks.inspectit.oce.core.utils.CommonUtils;
+import rocks.inspectit.oce.core.utils.StopWatch;
+
+import javax.annotation.PostConstruct;
+import java.lang.instrument.Instrumentation;
+import java.time.Duration;
+import java.util.*;
+
+/**
+ * This class is responsible for making sure that for every class the instrumentation
+ * eventually is applied as specified in the active {@link rocks.inspectit.oce.core.config.model.InspectitConfig}
+ * <p>
+ * Whenever a classes instrumentation configuration might have changes, this class gets put into this services working queue.
+ * This service is then responsible for filtering out the classes whose instrumentaiton actually has
+ * changed and submits them to the {@link ClassRetransformationService}.
+ */
+@Service
+@Slf4j
+public class InstrumentationUpdateService {
```

Comment:
I would like a uni test for this and an int test as well.. In the int test we can focus that interaction between services work, that configuration changes are applied, etc.. Maybe also the destruction of the context if possible..